src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/reader/ShardStateMetricsSnapshot.java

```diff
@@ -0,0 +1,286 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.reader;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.DBUtils;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metrics.AllMetrics;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.metricsdb.MetricsDB;
+import com.google.common.annotations.VisibleForTesting;
+
+import java.sql.Connection;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jooq.BatchBindStep;
+import org.jooq.DSLContext;
+import org.jooq.Field;
+import org.jooq.Record;
+import org.jooq.Result;
+import org.jooq.SQLDialect;
+import org.jooq.SelectField;
+import org.jooq.impl.DSL;
+
+public class ShardStateMetricsSnapshot implements Removable {
+    private static final Logger LOG = LogManager.getLogger(ShardStateMetricsSnapshot.class);
+    private final DSLContext create;
+    private final String tableName;
+    private static final Long EXPIRE_AFTER = 1200000L;
+    private List<Field<?>> columns;
+
+    public ShardStateMetricsSnapshot(Connection conn, Long windowStartTime) {
+        this.create = DSL.using(conn, SQLDialect.SQLITE);
+        this.tableName = "shard_state_" + windowStartTime;
+
+        this.columns =
+                new ArrayList<Field<?>>() {
```

Comment:
I am thinking, maybe we can statically load this as it is all constants ?

Reply:
Done